In [81]:
import pandas as pd
arrest = pd.read_csv("arrest_data.csv", sep=",")

In [82]:
arrest_dict = dict()
for index, row in arrest.iterrows():
    neighborhood_a = row['INCIDENTNEIGHBORHOOD']
    if neighborhood_a not in arrest_dict:
        arrest_dict[neighborhood_a] = 1
    else:
        arrest_dict[neighborhood_a] = arrest_dict[neighborhood_a] + 1
arrest_dict

{'Bloomfield': 930,
 'Outside City': 298,
 'Westwood': 209,
 'East Allegheny': 2014,
 'Crafton Heights': 546,
 'Greenfield': 497,
 'Brookline': 1268,
 'Regent Square': 35,
 'East Liberty': 1733,
 nan: 2326,
 'East Hills': 1084,
 'Carrick': 2199,
 'Allegheny Center': 1181,
 'South Side Flats': 3186,
 'Sheraden': 1192,
 'East Carnegie': 48,
 'Golden Triangle/Civic Arena': 83,
 'Point Breeze North': 261,
 'Perry South': 990,
 'Allentown': 884,
 'North Shore': 759,
 'Marshall-Shadeland': 1406,
 'Central Oakland': 488,
 'Middle Hill': 880,
 'Strip District': 472,
 'Lincoln-Lemington-Belmar': 1219,
 'Central Northside': 23,
 'Troy Hill-Herrs Island': 6,
 'Highland Park': 530,
 'Homewood North': 1837,
 'Northview Heights': 662,
 'Brighton Heights': 1025,
 'Hazelwood': 1287,
 'Perry North': 567,
 'Bedford Dwellings': 616,
 'Elliott': 619,
 'Beechview': 1116,
 'Fineview': 773,
 'Glen Hazel': 153,
 'Spring Hill-City View': 667,
 'Bluff': 1371,
 'Manchester': 445,
 'Homewood West': 732,
 'Outside

In [83]:
census = pd.read_csv("population-density.csv", sep=",")

In [84]:
census_dict = dict()
for index, row in census.iterrows():
    neighborhood_c = row['Neighborhood']
    if neighborhood_c not in census_dict:
        val = row['Pop. 2010']
        val = val.replace(',', '')
        census_dict[neighborhood_c] = int(val)
census_dict

{'Allegheny Center': 933,
 'Allegheny West': 462,
 'Allentown': 2500,
 'Arlington': 1869,
 'Arlington Heights': 244,
 'Banksville': 4144,
 'Bedford Dwellings': 1202,
 'Beechview': 7974,
 'Beltzhoover': 1925,
 'Bloomfield': 8442,
 'Bluff': 6600,
 'Bon Air': 808,
 'Brighton Heights': 7247,
 'Brookline': 13214,
 'California-Kirkbride': 761,
 'Carrick': 10113,
 'Central Business District': 3629,
 'Central Lawrenceville': 4482,
 'Central Northside': 2923,
 'Central Oakland': 6086,
 'Chartiers City': 477,
 'Chateau': 11,
 'Crafton Heights': 3814,
 'Crawford-Roberts': 2256,
 'Duquesne Heights': 2425,
 'East Allegheny': 2136,
 'East Carnegie': 570,
 'East Hills': 3169,
 'East Liberty': 5869,
 'Elliott': 2381,
 'Esplen': 301,
 'Fairywood': 1002,
 'Fineview': 1285,
 'Friendship': 1785,
 'Garfield': 3675,
 'Glen Hazel': 716,
 'Greenfield': 7294,
 'Hays': 362,
 'Hazelwood': 4317,
 'Highland Park': 6395,
 'Homewood North': 3280,
 'Homewood South': 2344,
 'Homewood West': 818,
 'Knoxville': 3747,
 '

In [89]:
arrest_to_pop_ratio = dict()
for key in census_dict:
        if key in arrest_dict:
            pop = census_dict[key]
            arrests = arrest_dict[key]
            ratio = arrests/pop
            arrest_to_pop_ratio[key] = ratio
arrest_to_pop_ratio

{'Allegheny Center': 1.2658092175777063,
 'Allegheny West': 0.26406926406926406,
 'Allentown': 0.3536,
 'Arlington': 0.14820759764579988,
 'Arlington Heights': 0.569672131147541,
 'Banksville': 0.06563706563706563,
 'Bedford Dwellings': 0.5124792013311148,
 'Beechview': 0.1399548532731377,
 'Beltzhoover': 0.24987012987012988,
 'Bloomfield': 0.11016346837242359,
 'Bluff': 0.20772727272727273,
 'Bon Air': 0.30816831683168316,
 'Brighton Heights': 0.14143783634607424,
 'Brookline': 0.09595883154230361,
 'California-Kirkbride': 0.3679369250985545,
 'Carrick': 0.21744289528329871,
 'Central Business District': 1.042987048773767,
 'Central Lawrenceville': 0.09259259259259259,
 'Central Northside': 0.00786862812179268,
 'Central Oakland': 0.0801840289188301,
 'Chartiers City': 0.08385744234800839,
 'Chateau': 37.72727272727273,
 'Crafton Heights': 0.14315679077084426,
 'Crawford-Roberts': 0.3404255319148936,
 'Duquesne Heights': 0.07463917525773196,
 'East Allegheny': 0.9428838951310862,
 'Ea

## Conclusion
---
After counting the number of arrests in each neighborhood, I used census data from 2010 to figure out the population of each neighborhood. If the neighborhood was included in both datasets, I found the ratio between the number of arrests and the population of each neighborhood. Whichever one of these ratios was the lowest will determine which neighborhood is the best to live in. I will list the top 5 for the sole purpose of comparing to the other submetrics that my group members have, here are the top 5 neighborhoods, from best to "worst".
1. Central Northside = 0.00786862812179268
2. Squirrel Hill North = 0.023761330634515535
3. Swisshelm Park = 0.029390154298310066
4. Regent Square = 0.03771551724137931
5. North Oakland = 0.03876409818974505